cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -477,6 +493,71 @@ static void sc_mount_nvidia_driver_multiarch(const char *rootfs_dir, const char
     }
 }
 
+static int sc_mount_exported_paths(const char *rootfs_dir) {
+    static const char *export_glob = {SC_SNAPD_EXPORT "/*.source"};
+
+    glob_t glob_res SC_CLEANUP(globfree) = {.gl_pathv = NULL};
+    int err = glob(export_glob, 0, NULL, &glob_res);
+    if (err == GLOB_NOMATCH) {
+        debug("no sources in export folder");
+        return 0;
+    }
+    if (err != 0) {
+        die("cannot search using glob pattern %s: %d", export_glob, err);
+    }
+    debug("sources in export folder, nvidia libraries from snaps");
+
+    // Create tmpfs
+    char tmpfs_path[PATH_MAX] = {0};
+    sc_must_snprintf(tmpfs_path, sizeof tmpfs_path, "%s%s", rootfs_dir, SC_LIBSNAP_DIR);
+    sc_mkdir_and_mount_tpmfs(tmpfs_path);
+
+    // Libs exported by snaps, bind mount them
+    char path[PATH_MAX] = {0};
+    char dest_path[PATH_MAX] = {0};
+    for (size_t i = 0; i < glob_res.gl_pathc; ++i) {
```

Comment:
My personal opinion on this for-loop needs some comments along the way to create better structure in this code, but it's not a blocker.